src/overlays/actors/ovl_Boss_Tw/z_boss_tw.c

```diff
@@ -156,183 +187,5337 @@ static ColliderCylinderInit D_8094A840 = {
+    }
+
+    if ((this->work[CS_TIMER_2] % 32) == 0) {
+        if (this->actionFunc != BossTw_TwinrovaMergeCS) {
+            if (Rand_ZeroOne() < 0.3f) {
+                this->work[BLINK_IDX] = 4;
+            }
+        }
+    }
+
+    if (this->actionFunc == BossTw_TwinrovaMergeCS) {
+        this->leftEyeTexIdx = D_8094A90C[this->work[TW_BLINK_IDX]];
+        if (this->work[TW_BLINK_IDX] != 0) {
+            this->work[TW_BLINK_IDX]--;
+        }
+    } else {
+        if (this->actionFunc == BossTw_TwinrovaStun) {
+            this->eyeTexIdx = 1;
+        }
+
+        if (this->actionFunc == BossTw_TwinrovaDeathCS) {
+            this->eyeTexIdx = 2;
+        }
+
+        this->leftEyeTexIdx = this->eyeTexIdx;
+    }
+
+    if (this->visible && this->unk_5F8 == 0) {
+        Vec3f pos;
+        Vec3f velocity = { 0.0f, 0.0f, 0.0f };
+        Vec3f accel;
+
+        if (this->work[UNK_S8] != 0) {
+            this->work[UNK_S8] -= 20;
+            if (this->work[UNK_S8] < 0) {
+                this->work[UNK_S8] = 0;
+            }
+        }
+
+        Math_ApproachF(&this->workf[UNK_F12], 1.0f, 1.0f, 0.05f);
+        accel.y = 0.4f;
+
+        for (i = 0; i < 2; i++) {
+            pos = this->leftScepterPos;
+            pos.x += Rand_CenteredFloat(30.0f);
+            pos.y += Rand_CenteredFloat(30.0f);
+            pos.z += Rand_CenteredFloat(30.0f);
+            accel.x = Rand_CenteredFloat(0.5f);
+            accel.z = Rand_CenteredFloat(0.5f);
+            BossTw_AddDotEffect(globalCtx, &pos, &velocity, &accel, (s16)Rand_ZeroFloat(2.0f) + 7, 0, 75);
+        }
+
+        for (i = 0; i < 2; i++) {
+            pos = this->rightScepterPos;
+            pos.x += Rand_CenteredFloat(30.0f);
+            pos.y += Rand_CenteredFloat(30.0f);
+            pos.z += Rand_CenteredFloat(30.0f);
+            accel.x = Rand_CenteredFloat(0.5f);
+            accel.z = Rand_CenteredFloat(0.5f);
+            BossTw_AddDotEffect(globalCtx, &pos, &velocity, &accel, (s16)Rand_ZeroFloat(2.0f) + 7, 1, 75);
+        }
+    }
+
+    this->collider.dim.radius = 35;
+
+    if (this->actionFunc == BossTw_TwinrovaSpin) {
+        this->collider.dim.radius *= 2;
+    }
+
+    this->collider.dim.height = 150;
+    this->collider.dim.yShift = -60;
+    Collider_UpdateCylinder(&this->actor, &this->collider);
+
+    if (this->work[INVINC_TIMER] == 0) {
+        if (this->actionFunc != BossTw_TwinrovaStun) {
+            if (this->twinrovaStun != 0) {
+                this->twinrovaStun = 0;
+                this->work[FOG_TIMER] = 10;
+                BossTw_TwinrovaDamage(this, globalCtx, 0);
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_TWINROBA_YOUNG_DAMAGE);
+            } else if (this->collider.base.acFlags & AC_HIT) {
+                ColliderInfo* info = this->collider.info.acHitInfo;
+                this->collider.base.acFlags &= ~AC_HIT;
+                if (info->toucher.dmgFlags & (DMG_SLINGSHOT | DMG_ARROW)) {}
```

Comment:
really

can this be replaced by a if(1){} / `if(info->toucher.dmgFlags){}` or does it actually do something as is

Reply:
has to be exactly this apparently lol